src/Language/Docker/PrettyPrint.hs

```diff
@@ -75,42 +82,43 @@ prettyPrintBaseImage BaseImage {..} = do
         Nothing -> mempty
         Just (Digest d) -> "@" <> pretty d
 
-prettyPrintPairs :: Pairs -> Doc ann
+prettyPrintPairs :: (?esc :: Char) => Pairs -> Doc ann
 prettyPrintPairs ps = align $ sepLine $ fmap prettyPrintPair ps
   where
-    sepLine = concatWith (\x y -> x <> " \\" <> line <> y)
+    sepLine = concatWith (\x y -> x <> " " <> pretty ?esc <> line <> y)
 
-prettyPrintPair :: (Text, Text) -> Doc ann
+prettyPrintPair :: (?esc :: Char) => (Text, Text) -> Doc ann
 prettyPrintPair (k, v) = pretty k <> pretty '=' <> doubleQoute v
 
-prettyPrintArguments :: Arguments Text -> Doc ann
+prettyPrintArguments :: (?esc :: Char) => Arguments Text -> Doc ann
 prettyPrintArguments (ArgumentsList as) = prettyPrintJSON (Text.words as)
 prettyPrintArguments (ArgumentsText as) = hsep (fmap helper (Text.words as))
   where
-    helper "&&" = "\\\n &&"
+    helper "&&" = pretty ?esc <> "\n &&"
     helper a = pretty a
 
-prettyPrintJSON :: [Text] -> Doc ann
+prettyPrintJSON :: (?esc :: Char) => [Text] -> Doc ann
 prettyPrintJSON args = list (fmap doubleQoute args)
 
-doubleQoute :: Text -> Doc ann
+doubleQoute :: (?esc :: Char) => Text -> Doc ann
 doubleQoute w = enclose dquote dquote (pretty (escapeQuotes w))
 
-escapeQuotes :: Text -> L.Text
+escapeQuotes :: (?esc :: Char) => Text -> L.Text
 escapeQuotes text =
   case Text.foldr accumulate (EscapeAccum mempty 0 False) text of
     EscapeAccum buffer _ False -> B.toLazyText buffer
     EscapeAccum buffer count True ->
       case count `mod` 2 of
-        0 -> B.toLazyText (B.singleton '\\' <> buffer)
+        0 -> B.toLazyText (B.singleton ?esc <> buffer)
         _ -> B.toLazyText buffer
   where
     accumulate '"' EscapeAccum {buffer, escaping = False} =
       EscapeAccum (B.singleton '"' <> buffer) 0 True
-    accumulate '\\' EscapeAccum {buffer, escaping = True, count} =
-      EscapeAccum (B.singleton '\\' <> buffer) (count + 1) True
+--    accumulate '\\' EscapeAccum {buffer, escaping = True, count} =
+--      EscapeAccum (B.singleton '\\' <> buffer) (count + 1) True
```

Comment:
do you want to delete this code already?

Reply:
Oops. Yes that should be deleted. The case in question is covered by the addition of a case in the patter below that line (line 118). Good spot.